### PR TITLE
Allowances API rework

### LINKED ISSUE #1064: Adapt api to handle allowance changes

### LINKED ISSUE #1105: Write Tests for new Allowance API

### PR DESCRIPTION
**Pull request by axrolld**
This PR separates construction, mareco, and linear allowance and builds two things :
- RJSAllowance is now either Engineering or Standard
- RJSAllowanceDistribution that is either MARECO or LINEAR

### CODE REVIEW

**Comment by alexandredamiron** [comment]
For @nicolaswurtz rebuild for new allowance logic core/api

**Review by flomonster — CHANGES_REQUESTED** [review]
A `.DS_Store` file was pushed by error

File: core/src/main/java/fr/sncf/osrd/railjson/parser/RJSStandaloneTrainScheduleParser.java [hunk 1]
```diff
@@ -69,52 +70,42 @@ private static Allowance parseAllowance(
...
-                            )
-                    )
-            );
+        var allowanceDistribution = rjsAllowance.distribution;
+        double beginPos;
+        double endPos;
+        List<AllowanceRange> ranges;
+        // parse allowance type
+        if (rjsAllowance instanceof RJSAllowance.EngineeringAllowance engineeringAllowance) {
+            beginPos = engineeringAllowance.beginPosition;
```
Comment (axrolld) [inline core/src/main/java/fr/sncf/osrd/railjson/parser/RJSStandaloneTrainScheduleParser.java:79]: Apparently these lines are not covered by tests, I'll right one to make code coverage work

File: front/public/locales/fr/allowances.json [hunk 2]
```diff
@@ -11,19 +11,30 @@
...
-    "construction": "Marche de construction modifiée"
+    "marecoAdd": "Marge ajoutée",
+    "marecoDel": "Marge supprimée",
+    "construction": "Marge modifiée"
   },
+  "specificValuesOnIntervals" : "Valeurs spécifiques sur intervales",
   "allowanceGlobalType": {
     "mareco": "Mareco",
-    "construction": "Construction"
+    "construction": "Construction",
```
Comment (axrolld) [inline front/public/locales/fr/allowances.json]: Probably not my scope to review front but i'm curious : why is there a `"construction": "Construction"` if there is a `"engineering": "Construction"` ?

File: front/public/locales/fr/allowances.json [hunk 3]
```diff
@@ -11,19 +11,30 @@
...
   "margedTrainSchedule": "Avec marges",
   "allowanceByInterval": "Marges par intervalles",
   "ecoAllowanceByInterval": "Marches Economiques par intervalles",
   "allowanceModified": {
-    "marecoAdd": "Marche économique ajoutée",
-    "marecoDel": "Marche économique supprimée",
-    "construction": "Marche de construction modifiée"
+    "marecoAdd": "Marge ajoutée",
+    "marecoDel": "Marge supprimée",
+    "construction": "Marge modifiée"
```
Comment (axrolld) [inline front/public/locales/fr/allowances.json]: I don't quite get this part, I think it is still quite confused between mareco, construction, etc.
For example why is construction corresponding to a modified allowance ?

**Review by nicolaswurtz — DISMISSED** [review]

**Review by nicolaswurtz — DISMISSED** [review]

**Review by flomonster — APPROVED** [review]

**Review by eckter — APPROVED** [review]

File: core/src/main/java/fr/sncf/osrd/railjson/schema/schedule/RJSAllowance.java [hunk 4]
```diff
@@ -6,56 +6,57 @@
 public class RJSAllowance {
     public static final PolymorphicJsonAdapterFactory<RJSAllowance> adapter = (
             PolymorphicJsonAdapterFactory.of(RJSAllowance.class, "allowance_type")
-                    .withSubtype(Construction.class, "construction")
-                    .withSubtype(Mareco.class, "mareco")
-                    .withSubtype(Linear.class, "linear")
+                    .withSubtype(EngineeringAllowance.class, "engineering")
+                    .withSubtype(StandardAllowance.class, "standard")
```
Comment (elzinko) [inline core/src/main/java/fr/sncf/osrd/railjson/schema/schedule/RJSAllowance.java:10]: I prefer using constants when using functional strings

**Review by elzinko — APPROVED** [review]
Big review but good job

**Review by elzinko — APPROVED** [review]

**Review by multun — APPROVED** [review]

**Review by nicolaswurtz — APPROVED** [review]

**Pull request merged**